### ISSUE TITLE
模型推理

### ISSUE THREAD

**Issue by xiaohui0225** [boschresearch/ISSA#1]
请问，可以直接用weights下的权重直接推理吗？推理的命令是什么呢？
config文件里有很多个文件路径，分别应该填写什么地址呢？
十分感谢

**Comment by YumengLi007** [comment]
Hi @xiaohui0225 ,

thanks for your interest in our work. The pretrained model could not be released unfortunately. But we provide the training scripts [train_encoder.py](https://github.com/SnowdenLee/ISSA/blob/main/train_encoder.py) and the [configs/mne_training.yml](https://github.com/SnowdenLee/ISSA/blob/main/configs/mne_training.yml) configuration file. 
The path in the config file represents:

- data: path to original Cityscapes
- data_fake: path to the fake images generated by the pretrained(frozen) GAN generator with ith w vector
- pkl_dir: path to the pretrained GAN generator

Please check [how-to.pdf](https://github.com/SnowdenLee/ISSA/blob/main/how-to.pdf) for more training details. 
For inference, it's relatively simple, you may refer to the logging code [here](https://github.com/SnowdenLee/ISSA/blob/a6d191f12f31c92608dd6e50f09d11113a8722b4/train_encoder.py#L428-L438). 

Please feel free to reach out if you have any further specific questions!

**Comment by xiaohui0225** [comment]
十分感谢您的回答，我是一个小白，还有以下疑问想请教一下您：
1、daka_fake和pkl_dir 分别是什么呢？您的解释我没有搞明白，这两个应该怎么获取呢？或者说是怎么生成的呢？
2、ISSA的训练指令是什么呢？如果训练好了，推理的指令又是什么呢？
3、在您的代码里有个`ISSA/tree/main/training/lpips/weights
/v0.0/`，里面有三个pth文件，请问这三个权重有什么用呢？

**Comment by YumengLi007** [comment]
1. This repo only provides training code for the Masked Noise Encoder. For the GAN generator training, please refer to [stylegan3](https://github.com/NVlabs/stylegan3). Given the pretrained GAN generator, we generate some fake images, e.g., 50K images, depending on the size of your dataset. You can sample Z to generate images, correspondingly we can also have the style vector w, which is the output of the mapping network of the generator. We store both image and its corresponding style w vector. ```data_fake``` refers to the path where you store these.   
```pkl_dir``` is the path where you store the pretrained GAN generator, where for example you could train using [stylegan3](https://github.com/NVlabs/stylegan3) or use their provided checkpoints depending on dataset you would like to use.

2. you could simply run ``` python train_encoder.py ```  and configure the path and parameters in [configs/mne_training.yml](https://github.com/SnowdenLee/ISSA/blob/main/configs/mne_training.yml).   
For reference, we don't have a dedicated script in this repo. But you could check the logging code [here](https://github.com/SnowdenLee/ISSA/blob/a6d191f12f31c92608dd6e50f09d11113a8722b4/train_encoder.py#L428-L438), which show how the Encoder & Generator are used for image generation.

3.  These weights are for calculating the LPIPS loss

**Comment by xiaohui0225** [comment]
Thanks.
According to what you said, the train_encoder.py code is used to train the model, so how to infer after training? What are the running instructions?

**Comment by YumengLi007** [comment]
1. You firstly need to load the generator & encoder, similar to [here](https://github.com/boschresearch/ISSA/blob/d5d30bb813cb08bcaa9415e9c09b77182765a934/train_encoder.py#L207-L213)
2.  Then you can use the encoder and generator like [here](https://github.com/SnowdenLee/ISSA/blob/a6d191f12f31c92608dd6e50f09d11113a8722b4/train_encoder.py#L428-L438) to generate image. From a high level, you need to use the encoder to generate the style and noise, then pass them to the generator. 
There is no separate script for inference in this repo. You may need to combine the code snippets mentioned above for inference.

**Comment by YumengLi007** [comment]
Closed for now. Please feel free to reopen if there is anything still unclear.
